Automattic/woocommerce-payments
# Add "Needs Response" disputes page filter option

Partially fixes #4124

#### Changes proposed in this Pull Request





* A "Needs response" filter is added to the "Show" dropdown on the **Payments → Disputes** page.
* When selected, disputes with the status `needs_response` or `warning_needs_response` are displayed.
* This filter can be selected on page load if the `filter=awaiting_response` query param is provided.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/3147296/174949554-bab1f951-ccc1-4f1c-9928-b4da918bfce1.png">

Filtering by multiple dispute `status` values is achieved with the WCPay-server disputes API `search` query param added in 2127-gh-Automattic/woocommerce-payments-server





#### Testing instructions





1. Checkout the WCPay-server branch: `add/2124-disputes-endpoints-allow-filtering-by-multiple-status` (2127-gh-Automattic/woocommerce-payments-server) and start the local WCPay-server environment with `npm start`.
2. Checkout this WCPay client branch and build JS bundles with `npm run build:client` or `npm start`
3. In your local WCPay (client) store, create disputed purchases with various statuses, including [`needs_response`, `warning_needs_response`, `lost`]:
   * Create a `warning_needs_response` disputed purchase with the card: `4000000000005423`.
   * Create a `needs_response` disputed purchase with the card: `4000000000000259`.
   * Create a `lost` disputed purchase with the card: `4000000000000259` and **Accept** the dispute as merchant.
4. View the **Payments → Disputes** page in wp-admin.
5. The "Show" dropdown should have "All disputes" selected and include the "Needs response" option at the top:
   <img width="400" alt="image" src="https://user-images.githubusercontent.com/3147296/174949554-bab1f951-ccc1-4f1c-9928-b4da918bfce1.png">
6. Selecting the "Needs response" option will:
   - trigger new network GET requests: `disputes` and `disputes/summary`
   - once resolved, the disputes table will only show disputes needing a response (*Needs Response* or *Inquiry: Needs Response*)
7. Manually **removing** the `filter=awaiting_response` query param from the browser URL bar and hitting `return` will select the "All disputes" filter on page load.
8. Manually **adding** the `filter=awaiting_response` query param to the browser URL bar and hitting `return` will select the "Needs response filter on page load.

-------------------


**Post merge**
